server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -59,15 +59,16 @@ pub(crate) fn generate(
         .tax
         .map(|tax| tax.percentage)
         .unwrap_or(update_invoice.tax_percentage);
-    update_invoice.default_donor_id = input_donor_id.clone().or(update_invoice.default_donor_id);
+    update_invoice.default_donor_id = input_donor_id.clone();
 
     if let Some(status) = patch.status.clone() {
         update_invoice.status = status.full_status()
     }
 
     if let Some(other_party) = other_party_option {
         update_invoice.name_store_id = other_party.store_id().map(|id| id.to_string());
-        // TODO: Should be using the name link id not the name row id...
+        // Assigning name_row id as name_link is ok, input name_row should always an active name
+        // - only querying needs to go via link table
```

Comment:
What if it goes through many merge iterations and the name_link_id is no longer the name_row id? 👀

Reply:
See linked comment and readme here - https://github.com/msupply-foundation/open-msupply/issues/7429

Name link records should always exist for every name that was, just might end up pointing to a different actual name upon querying again.